research/gnn-survey-paper/train.py

```diff
@@ -0,0 +1,98 @@
+import os
+import time
+import numpy as np
+import tensorflow as tf
+
+from absl import app
+from absl import flags
+
+from utils import load_dataset, build_model, cal_acc
+
+flags.DEFINE_string('dataset', 'cora',
+                    'The input dataset. Avaliable dataset now: cora')
+flags.DEFINE_string('model', 'gcn',
+                    'GNN model. Available model now: gcn')
+flags.DEFINE_float('dropout', 0.5, 'Dropout probability')
+flags.DEFINE_integer('gpu', '-1', 'Gpu id, -1 means cpu only')
+flags.DEFINE_float('lr', 1e-2, 'Initial learning rate')
+flags.DEFINE_integer('epochs', 200, 'Number of training epochs')
+flags.DEFINE_integer('num_layers', 2, 'Number of gnn layers')
+flags.DEFINE_list('hidden_dim', [32], 'Dimension of gnn hidden layers')
+flags.DEFINE_string('optimizer', 'adam', 'Optimizer for training')
```

Comment:
You can use DEFINE_enum when there are only a finite number of choices.

Reply:
OK!
I changed dataset, models, optimizer to DEFINE_enum type.